Assets/Character/Scripts/Consumable.cs

```diff
@@ -0,0 +1,49 @@
+﻿using UnityEngine;
+using System.Collections;
+
+public class Consumable : MonoBehaviour {
+
+    private int count;
+    private float cooldown;
+    private float startTime;
+    private bool healing;
+    private float duration;
+    private float interval;
+    private float timer;
+
+	// public Consumable?
+	void Start () {
+        count = Constants.CONSUMABLE_COUNT;
+        cooldown = Constants.CONSUMABLE_CD;
+        healing = false;
+        duration = Constants.CONSUMABLE_DURATION;
+        timer = 0;
+	}
+	
+	void Update () {
+        if(healing)
+        {
+            if (Time.time - startTime > duration)
+            {
+                healing = false;
+            }
+            else
+            {
+                timer += Time.deltaTime;
+                if (timer > interval)
+                {
+                    //add hp
+                    timer = 0;
+                }
+            }
+        }
+	    else if (PlayerInput.PressedConsumable() && (Time.time - startTime > duration) && count > 0)
```

Comment:
Just expose the .consume() function mentioned above to the player. Then the player will call consume if PressedConsumble() is true. Consumable shouldn't handle any input.